### PR TITLE
Add cfg! check for fake_crypto

### LINKED ISSUE #632: Disabling validator_block_post test when using fake_crypto
## Description

Currently, the test `validator_block_post` in the `rest_api` crate fails when running with `cargo test --all --all-features`

## Expected Behaviour

We'd like this test to pass when using the aforementioned tests

## Steps to resolve

Disabling the two failing asserts when when using the `fake-crypto` feature should suffice.

### PR DESCRIPTION
**Pull request by pscott**
## Issue Addressed

Closes #632 

## Proposed Changes

Adding cfg! test for fake_crypto in `validator_block_post` test.

### CODE REVIEW

File: beacon_node/rest_api/tests/test.rs [hunk 1]
```diff
@@ -338,11 +341,14 @@ fn validator_block_post() {
         .runtime()
         .block_on(remote_node.http.validator().publish_block(block.clone()))
         .expect("should publish block");
-    assert_eq!(
-        publish_status,
-        PublishStatus::Valid,
-        "the signed published block should be valid"
-    );
+
+    if cfg!(feature = "fake_crypto") {
```
Comment (michaelsproul) [inline beacon_node/rest_api/tests/test.rs]: If the function is marked `ignore` when `fake_crypto` is enabled, then disabling the asserts is redundant. I think we should remove the function ignore attribute and just ignore the asserts (although I think `cfg!(not(feature = "fake_crypto"))` is probably what you meant)

Comment (pscott) [reply beacon_node/rest_api/tests/test.rs]: Hehe of course! My bad on this one, I thought I had already removed the function attribute... I didn't review my own PR, bad Scott!

**Comment by michaelsproul** [comment]
Oh I see it has already passed here: https://github.com/pscott/lighthouse/commit/997e742a482487e5286f5571a343cac1f981c587/checks?check_suite_id=328992861

**Comment by pscott** [comment]
Yeah as Paul mentioned, it is a known issue and github is expected to have this solved before the end of the year! :D

**Comment by pscott** [comment]
So I must say I'm definitely not an expert on how features work.... I was just running my regular `cargo est --all --all-features` and having this test fail, regardless on which branch I was on. When I asked Paul for some help he pointed out this `fake_crypto` feature, and indeed using cfg! I managed to have it working... ^^'

**Pull request merged**